lib/src/command/lish.dart

```diff
@@ -251,64 +271,120 @@ the \$PUB_HOSTED_URL environment variable.''',
       usageException('Cannot use both --force and --dry-run.');
     }
 
-    if (entrypoint.root.pubspec.isPrivate) {
+    if (_fromArchive != null && _toArchive != null) {
+      usageException('Cannot use both --from-archive and --to-archive.');
+    }
+
+    if (_fromArchive != null && dryRun) {
+      usageException('Cannot use both --from-archive and --dry-run.');
+    }
+
+    if (_toArchive != null && dryRun) {
+      usageException('Cannot use both --to-archive and --dry-run.');
+    }
+
+    if (_fromArchive != null && force) {
+      usageException('Cannot use both --from-archive and --force.');
+    }
+
+    if (_toArchive != null && force) {
+      usageException('Cannot use both --to-archive and -force.');
+    }
+
+    if (_fromArchive == null && entrypoint.root.pubspec.isPrivate) {
       dataError('A private package cannot be published.\n'
           'You can enable this by changing the "publish_to" field in your '
           'pubspec.');
     }
```

Comment:
Hmm, isn't it wrong to access `entrypoint`? I mean it reads from current directory, which is always wrong when `--from-archive` is given.

We should assume that CWD is not a package, and certainly not the package that you want to publish.

Reply:
We only access the entrypoint when `_fromArchive` is null. Lazy eval etc.